Code/GraphMol/FileParsers/testMultithreadedMolSupplier.cpp

```diff
@@ -0,0 +1,166 @@
+//
+//  Copyright (C) 2020 Shrey Aryan
+//
+//   @@ All Rights Reserved @@
+//  This file is part of the RDKit.
+//  The contents are covered by the terms of the BSD license
+//  which is included in the file license.txt, found at the root
+//  of the RDKit source tree.
+//
+#include <GraphMol/MonomerInfo.h>
+#include <GraphMol/RDKitBase.h>
+#include <GraphMol/SmilesParse/SmilesWrite.h>
+#include <RDGeneral/ConcurrentQueue.h>
+#include <RDGeneral/FileParseException.h>
+#include <RDGeneral/RDLog.h>
+#include <RDGeneral/test.h>
+
+#include <boost/dynamic_bitset.hpp>
+#include <boost/iostreams/device/file.hpp>
+#include <boost/iostreams/filtering_stream.hpp>
+#include <chrono>
+#include <fstream>
+#include <iomanip>
+#include <iostream>
+#include <map>
+#include <memory>
+#include <sstream>
+#include <string>
+
+#include "MultithreadedMolSupplier.h"
+#include "MultithreadedSmilesMolSupplier.h"
+
+namespace io = boost::iostreams;
+using namespace RDKit;
+using namespace std::chrono;
+
+struct PrintThread : public std::stringstream {
+  static inline std::mutex cout_mutex;
+  ~PrintThread() {
+    std::lock_guard<std::mutex> l{cout_mutex};
+    std::cout << rdbuf();
+  }
+};
+
+void testSmiConcurrent(std::string path, std::string delimiter,
+                       int smilesColumn, int nameColumn, bool titleLine,
+                       bool sanitize, int numWriterThreads,
+                       size_t sizeInputQueue, size_t sizeOutputQueue,
+                       unsigned int expectedResult) {
+  std::string rdbase = getenv("RDBASE");
+  std::string fname = rdbase + path;
+  unsigned int nMols = 0;
+  boost::dynamic_bitset<> bitVector(expectedResult);
+  MultithreadedSmilesMolSupplier sup(fname, delimiter, smilesColumn, nameColumn,
+                                     titleLine, sanitize, numWriterThreads,
+                                     sizeInputQueue, sizeOutputQueue);
+  //! we have not called the next method yet
+  TEST_ASSERT(sup.getLastRecordId() == 0);
+  //! initially no bit is set in the bitVector, sanity check
+  TEST_ASSERT(!bitVector.any());
+
+  while (!sup.atEnd()) {
+    ROMol* mol = sup.next();
+    unsigned int id = sup.getLastRecordId();
+    bitVector[id - 1] = 1;
+    if (mol != nullptr) {
+      ++nMols;
+    }
+    delete mol;
+  }
+  //! if all bits are set then we have seen possible ids
+  TEST_ASSERT(bitVector.all());
+  TEST_ASSERT(nMols == expectedResult);
+}
+
+void testSmiOld(std::string path, std::string delimiter, int smilesColumn,
```

Comment:
do you still need this function?

Reply:
To test performance in the method `testSmiPerformance()` I call this method to compare execution times between the concurrent and non-concurrent Smiles Mol Supplier. Should I remove the performance testing method, then I can also get of the function `testSmiOld(..)`?